module/zfs/vdev.c

```diff
@@ -898,6 +898,16 @@ vdev_alloc(spa_t *spa, vdev_t **vdp, nvlist_t *nv, vdev_t *parent, uint_t id,
 	(void) nvlist_lookup_uint64(nv, ZPOOL_CONFIG_CREATE_TXG,
 	    &vd->vdev_crtxg);
 
+	if (vd->vdev_ops == &vdev_root_ops &&
+	    (alloctype == VDEV_ALLOC_LOAD ||
+	    alloctype == VDEV_ALLOC_SPLIT ||
+	    alloctype == VDEV_ALLOC_ROOTPOOL)) {
+		(void) nvlist_lookup_uint64(nv,
+		    ZPOOL_CONFIG_VDEV_ROOT_ZAP, &spa->spa_root_vdev_zap);
+	} else {
+		ASSERT0(spa->spa_root_vdev_zap);
```

Comment:
This assertion tripped in ztest:

```
Executing /sbin/zdb -bccsv -G -d -Y -e -y  -p /var/tmp/zloop-run ztest
spa->spa_root_vdev_zap == 0 (0x42 == 0)
ASSERT at module/zfs/vdev.c:908:vdev_alloc()Aborted (core dumped)
ztest: '/sbin/zdb -bccsv -G -d -Y -e -y  -p /var/tmp/zloop-run ztest' exit code 134
```

Reply:
Thanks, I'll update this PR when I get a moment to look at this.

Reply:
@rob-wing Just in case you were unaware (since it is a pain to find them), there are artifacts that can be downloaded from here that might or might not be useful in debugging:

https://github.com/openzfs/zfs/actions/runs/4117171010

Reply:
We'll want to drop this ASSERT based on the conversation above, https://github.com/openzfs/zfs/pull/14405#discussion_r1122407870